src/reflect/scala/reflect/internal/tpe/CommonOwners.scala

```diff
@@ -27,7 +27,9 @@ private[internal] trait CommonOwners {
   protected def commonOwnerMap: CommonOwnerMap = commonOwnerMapObj
```

Comment:
this is the only usage of CommonOwnerMap
```
protected[internal] def commonOwner(tps: List[Type]): Symbol = {
    if (tps.isEmpty) NoSymbol
    else {
      commonOwnerMap.clear()
      tps foreach (commonOwnerMap traverse _)
      if (commonOwnerMap.result ne null) commonOwnerMap.result else NoSymbol
    }
  }
```
This looks more like local storage, or a pool. That would also allow removal of the override, which is just ensuring thatthere is no sharing